pkg/core/experiment.go

```diff
@@ -58,5 +61,33 @@ type Experiment struct {
 	Action         string `json:"action"`
 	RecoverCommand string `json:"recover_command"`
 
-	Cron string `json:"cron"`
+	cachedRequestCommand AttackConfig
+}
+
+func (exp *Experiment) GetRequestCommand() (AttackConfig, error) {
+	if exp.cachedRequestCommand != nil {
+		return exp.cachedRequestCommand, nil
+	}
+
+	var attackConfig AttackConfig
+	switch exp.Kind {
```

Comment:
I would suggest put this in a separate `func getCommandByKind(kind string) AttackConfig` and put it in some `utils.go` for reuse.

Reply:
@AsterNighT hmmm.. the use case is unmarshalling here so using interface `AttackConfig` works. In other use-cases, I can't imagine a utility function which just returns an interface, and without even deserialized data, would be reusable. Since we'd still need to do this switch case again to differentiate between which of the concrete types the interface belongs to.